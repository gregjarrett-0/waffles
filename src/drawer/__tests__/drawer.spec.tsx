/* eslint-disable @typescript-eslint/no-empty-function */
import { render, fireEvent, waitFor, act } from '@testing-library/react';

import { Drawer } from '../index';

const MOCKED_ID = '123abC';

jest.mock('../../icon', () => {
  return {
    Cross: () => 'CrossIcon',
  };
});

jest.mock('nanoid', () => {
  return {
    nanoid: () => MOCKED_ID,
  };
});

type DrawerTestProps = {
  isOpen: boolean;
  onClose: () => void;
  placement: 'left' | 'right';
};

function DrawerTest({ isOpen, onClose, placement }: DrawerTestProps) {
  return (
    <Drawer
      isOpen={isOpen}
      onClose={onClose}
      placement={placement}
      data-testid="test-drawer"
    >
      <Drawer.Header>Fabulous Marillion concert</Drawer.Header>
      <Drawer.Body>Write a script for a jesters tear yourself.</Drawer.Body>
      <Drawer.Footer>
        <Drawer.Button onClick={onClose}>Dismiss</Drawer.Button>
        <Drawer.Button variant="primary" autoFocus>
          Confirm
        </Drawer.Button>
      </Drawer.Footer>
    </Drawer>
  );
}

describe('Drawer', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('renders drawer and all subcomponents with all a11y attributes passed', async () => {
    const { getByTestId, getByText, getByLabelText } = render(
      <DrawerTest isOpen={true} onClose={() => {}} placement="left" />,
    );

    // Let slide in animations finish
    act(() => {
      jest.advanceTimersByTime(500);
    });

    let overlay;
    await waitFor(() => {
      overlay = getByTestId('modal-overlay');
    });
    const dialog = getByTestId('test-drawer');
    const header = getByText('Fabulous Marillion concert');
    const body = getByText(/script for a jesters tear/i);
    const closeButton = getByLabelText('Close');
    const confirmButton = getByText('Confirm');

    expect(dialog).toBeInTheDocument();
    expect(dialog).toHaveAttribute(
      'aria-labelledby',
      `modal-${MOCKED_ID}-header`,
    );
    expect(dialog).toHaveAttribute(
      'aria-describedby',
      `modal-${MOCKED_ID}-body`,
    );
    expect(overlay).toBeInTheDocument();
    expect(header).toBeInTheDocument();
    expect(header).toHaveAttribute('id', `modal-${MOCKED_ID}-header`);
    expect(body).toBeInTheDocument();
    expect(body).toHaveAttribute('id', `modal-${MOCKED_ID}-body`);
    expect(body).toBeInTheDocument();
    expect(closeButton).toBeInTheDocument();
    expect(confirmButton).toBeInTheDocument();
  });

  it('when custom close button is provided, appropriate onClose and onClick handlers are fired', async () => {
    jest.useFakeTimers();

    const handleClick = jest.fn();
    const handleClose = jest.fn();
    const { getByTestId } = render(
      <Drawer
        isOpen={true}
        onClose={handleClose}
        placement="left"
        closeButtonOverride={
          <Drawer.CloseButton
            data-testid="custom-close-button"
            onClick={handleClick}
          />
        }
      >
        <Drawer.Header>Fabulous Marillion concert</Drawer.Header>
        <Drawer.Body>Write a script for a jesters tear yourself.</Drawer.Body>
        <Drawer.Footer>
          <Drawer.Button variant="primary" onClick={handleClose}>
            Dismiss
          </Drawer.Button>
        </Drawer.Footer>
      </Drawer>,
    );

    let closeButton;
    await waitFor(() => {
      closeButton = getByTestId('custom-close-button');
    });

    expect(closeButton).toBeInTheDocument();
    closeButton && fireEvent.click(closeButton);

    // Component unmount is delayed because of close animation
    act(() => {
      jest.advanceTimersByTime(1000);
    });

    // Check default `onClose` behavior still works
    expect(handleClose).toHaveBeenCalledTimes(1);
    // Check custom close button `onClick` has also been called
    expect(handleClick).toHaveBeenCalledTimes(1);

    jest.useRealTimers();
  });

  it("doesn't render drawer when it is closed", () => {
    const { queryByTestId } = render(
      <DrawerTest isOpen={true} onClose={() => {}} placement="left" />,
    );

    const overlay = queryByTestId('modal-overlay');
    const dialog = queryByTestId('test-drawer');

    expect(dialog).not.toBeInTheDocument();
    expect(overlay).not.toBeInTheDocument();
  });

  it('when close button at the top is clicked, onClose handler is called', async () => {
    const handleClose = jest.fn();
    const { getByLabelText } = render(
      <DrawerTest isOpen={true} onClose={handleClose} placement="left" />,
    );

    // Let slide in animations finish
    act(() => {
      jest.advanceTimersByTime(500);
    });

    let closeButton;
    await waitFor(() => {
      closeButton = getByLabelText('Close');
    });

    closeButton && fireEvent.click(closeButton);

    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('when dismiss button in the footer is clicked, onClose handler is called', async () => {
    const handleClose = jest.fn();
    const { getByText } = render(
      <DrawerTest isOpen={true} onClose={handleClose} placement="left" />,
    );

    // Let slide in animations finish
    act(() => {
      jest.advanceTimersByTime(500);
    });

    let dismissButton;
    await waitFor(() => {
      dismissButton = getByText('Dismiss');
    });

    dismissButton && fireEvent.click(dismissButton);

    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('when drawer is open, button with autoFocus flag set should be focused', async () => {
    const { getByText } = render(
      <DrawerTest isOpen={true} onClose={() => {}} placement="left" />,
    );

    // Let slide in animations finish
    act(() => {
      jest.advanceTimersByTime(500);
    });

    let confirmButton;
    await waitFor(() => {
      confirmButton = getByText('Confirm').closest('button');
    });

    expect(confirmButton).toHaveFocus();
  });

  it('renders snapshot when drawer is placed to the left', async () => {
    const { getByTestId } = render(
      <DrawerTest isOpen={true} onClose={() => {}} placement="left" />,
    );

    // Let slide in animations finish
    act(() => {
      jest.advanceTimersByTime(500);
    });

    let drawer;
    await waitFor(() => {
      drawer = getByTestId('test-drawer');
    });

    expect(drawer).toMatchSnapshot();
  });

  it('renders snapshot when drawer is placed to the right', async () => {
    const { getByTestId } = render(
      <DrawerTest isOpen={true} onClose={() => {}} placement="right" />,
    );

    // Let slide in animations finish
    act(() => {
      jest.advanceTimersByTime(500);
    });

    let drawer;
    await waitFor(() => {
      drawer = getByTestId('test-drawer');
    });

    expect(drawer).toMatchSnapshot();
  });
});
