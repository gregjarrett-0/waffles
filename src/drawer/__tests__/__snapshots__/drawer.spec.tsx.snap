// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders snapshot when drawer is placed to the left 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translateY(12px);
    -moz-transform: translateY(12px);
    -ms-transform: translateY(12px);
    transform: translateY(12px);
  }

  to {
    opacity: 1;
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

@keyframes animation-1 {
  from {
    -webkit-transform: translateX(-356px);
    -moz-transform: translateX(-356px);
    -ms-transform: translateX(-356px);
    transform: translateX(-356px);
  }

  to {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }
}

@keyframes animation-2 {
  from {
    -webkit-transform: translateX(-600px);
    -moz-transform: translateX(-600px);
    -ms-transform: translateX(-600px);
    transform: translateX(-600px);
  }

  to {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }
}

.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 180px;
  max-height: 100%;
  min-width: 100%;
  max-width: 600px;
  background-color: #FFFFFF;
  border-radius: 4px;
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  outline: 0;
  overflow: hidden;
  pointer-events: all;
  opacity: 0;
  -webkit-transform: translateY(12px);
  -moz-transform: translateY(12px);
  -ms-transform: translateY(12px);
  transform: translateY(12px);
  -webkit-animation: animation-0 200ms cubic-bezier(0.2, 0.8, 0.6, 1) forwards;
  animation: animation-0 200ms cubic-bezier(0.2, 0.8, 0.6, 1) forwards;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  height: 100vh;
  z-index: 9000;
  opacity: 1;
  width: calc(100vw - 64px);
  max-width: 356;
  -webkit-transform: translateX(
        -356px
      );
  -moz-transform: translateX(
        -356px
      );
  -ms-transform: translateX(
        -356px
      );
  transform: translateX(
        -356px
      );
  -webkit-animation: animation-1 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
  animation: animation-1 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
}

@media screen and (min-width: 480px) {
  .emotion-0 {
    min-width: 400px;
  }
}

@media screen and (min-width: 480px) {
  .emotion-0 {
    width: min(
            calc(100vw - 64px),
            600px
          );
    max-width: min(
            calc(100vw - 64px),
            600px
          );
    -webkit-transform: translateX(
            -600px
          );
    -moz-transform: translateX(
            -600px
          );
    -ms-transform: translateX(
            -600px
          );
    transform: translateX(
            -600px
          );
    -webkit-animation: animation-2 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
    animation: animation-2 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
  }
}

.emotion-1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #0075AD;
  background-color: transparent;
  border-color: transparent;
  position: absolute;
  top: 8px;
  right: 8px;
  z-index: 1;
  color: #05192D;
}

.emotion-1::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-1:hover:not(:disabled) {
  background-color: rgba(5, 25, 45, 0.1);
}

.emotion-1:active:not(:disabled) {
  background-color: transparent;
}

.emotion-1:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-2 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  padding-top: 24px;
  padding-right: 48px;
  padding-bottom: 16px;
  padding-left: 24px;
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-3 {
  padding-top: 0;
  padding-right: 24px;
  padding-bottom: 8px;
  padding-left: 24px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow-x: hidden;
  overflow-y: auto;
}

.emotion-4 {
  padding: 16px 24px;
  border-top: 1px solid rgba(5, 25, 45, 0.15);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 8px;
}

.emotion-5 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.emotion-5::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-5:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-5:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-5:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-5:first-of-type {
  margin-right: auto;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.emotion-7::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-7:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-7:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-7:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-7::after {
  box-shadow: 0 0 0 2px #009BD8;
}

.emotion-7:active:focus {
  background-color: #00D649;
}

.emotion-7:first-of-type {
  margin-right: auto;
}

<section
  aria-describedby="drawer-123abC-body"
  aria-labelledby="drawer-123abC-header"
  aria-modal="true"
  class="emotion-0"
  data-testid="test-drawer"
  role="dialog"
  tabindex="-1"
>
  <button
    aria-label="Close"
    class="emotion-1"
    type="button"
  >
    CrossIcon
  </button>
  <h2
    class="emotion-2"
    id="drawer-123abC-header"
  >
    Fabulous Marillion concert
  </h2>
  <div
    class="emotion-3"
    id="drawer-123abC-body"
  >
    Write a script for a jesters tear yourself.
  </div>
  <footer
    class="emotion-4"
  >
    <button
      class="emotion-5"
      type="button"
    >
      <span
        class="emotion-6"
      >
        Dismiss
      </span>
    </button>
    <button
      class="emotion-7"
      type="button"
    >
      <span
        class="emotion-6"
      >
        Confirm
      </span>
    </button>
  </footer>
</section>
`;

exports[`Drawer renders snapshot when drawer is placed to the right 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translateY(12px);
    -moz-transform: translateY(12px);
    -ms-transform: translateY(12px);
    transform: translateY(12px);
  }

  to {
    opacity: 1;
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translateY(12px);
    -moz-transform: translateY(12px);
    -ms-transform: translateY(12px);
    transform: translateY(12px);
  }

  to {
    opacity: 1;
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }
}

@keyframes animation-1 {
  from {
    -webkit-transform: translateX(356px);
    -moz-transform: translateX(356px);
    -ms-transform: translateX(356px);
    transform: translateX(356px);
  }

  to {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }
}

@keyframes animation-2 {
  from {
    -webkit-transform: translateX(600px);
    -moz-transform: translateX(600px);
    -ms-transform: translateX(600px);
    transform: translateX(600px);
  }

  to {
    -webkit-transform: translateX(0);
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }
}

.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 180px;
  max-height: 100%;
  min-width: 100%;
  max-width: 600px;
  background-color: #FFFFFF;
  border-radius: 4px;
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  outline: 0;
  overflow: hidden;
  pointer-events: all;
  opacity: 0;
  -webkit-transform: translateY(12px);
  -moz-transform: translateY(12px);
  -ms-transform: translateY(12px);
  transform: translateY(12px);
  -webkit-animation: animation-0 200ms cubic-bezier(0.2, 0.8, 0.6, 1) forwards;
  animation: animation-0 200ms cubic-bezier(0.2, 0.8, 0.6, 1) forwards;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  height: 100vh;
  z-index: 9000;
  opacity: 1;
  width: calc(100vw - 64px);
  max-width: 356;
  -webkit-transform: translateX(
        356px
      );
  -moz-transform: translateX(
        356px
      );
  -ms-transform: translateX(
        356px
      );
  transform: translateX(
        356px
      );
  -webkit-animation: animation-1 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
  animation: animation-1 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
}

@media screen and (min-width: 480px) {
  .emotion-0 {
    min-width: 400px;
  }
}

@media screen and (min-width: 480px) {
  .emotion-0 {
    width: min(
            calc(100vw - 64px),
            600px
          );
    max-width: min(
            calc(100vw - 64px),
            600px
          );
    -webkit-transform: translateX(
            600px
          );
    -moz-transform: translateX(
            600px
          );
    -ms-transform: translateX(
            600px
          );
    transform: translateX(
            600px
          );
    -webkit-animation: animation-2 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
    animation: animation-2 200ms cubic-bezier(0, 0, 0.6, 1) forwards;
  }
}

.emotion-1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #0075AD;
  background-color: transparent;
  border-color: transparent;
  position: absolute;
  top: 8px;
  right: 8px;
  z-index: 1;
  color: #05192D;
}

.emotion-1::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-1:hover:not(:disabled) {
  background-color: rgba(5, 25, 45, 0.1);
}

.emotion-1:active:not(:disabled) {
  background-color: transparent;
}

.emotion-1:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-2 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  padding-top: 24px;
  padding-right: 48px;
  padding-bottom: 16px;
  padding-left: 24px;
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-3 {
  padding-top: 0;
  padding-right: 24px;
  padding-bottom: 8px;
  padding-left: 24px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow-x: hidden;
  overflow-y: auto;
}

.emotion-4 {
  padding: 16px 24px;
  border-top: 1px solid rgba(5, 25, 45, 0.15);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 8px;
}

.emotion-5 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.emotion-5::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-5:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-5:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-5:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-5:first-of-type {
  margin-right: auto;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.emotion-7::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-7:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-7:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-7:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-7::after {
  box-shadow: 0 0 0 2px #009BD8;
}

.emotion-7:active:focus {
  background-color: #00D649;
}

.emotion-7:first-of-type {
  margin-right: auto;
}

<section
  aria-describedby="drawer-123abC-body"
  aria-labelledby="drawer-123abC-header"
  aria-modal="true"
  class="emotion-0"
  data-testid="test-drawer"
  role="dialog"
  tabindex="-1"
>
  <button
    aria-label="Close"
    class="emotion-1"
    type="button"
  >
    CrossIcon
  </button>
  <h2
    class="emotion-2"
    id="drawer-123abC-header"
  >
    Fabulous Marillion concert
  </h2>
  <div
    class="emotion-3"
    id="drawer-123abC-body"
  >
    Write a script for a jesters tear yourself.
  </div>
  <footer
    class="emotion-4"
  >
    <button
      class="emotion-5"
      type="button"
    >
      <span
        class="emotion-6"
      >
        Dismiss
      </span>
    </button>
    <button
      class="emotion-7"
      type="button"
    >
      <span
        class="emotion-6"
      >
        Confirm
      </span>
    </button>
  </footer>
</section>
`;
